From 9b4e9f5abb94b671f998ffc0efebd1582852fee3 Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Thu, 26 Jul 2018 12:17:40 +0800
Subject: [PATCH] bcache: do not assign in if condition in bcache_device_init()
Git-commit: 9b4e9f5abb94b671f998ffc0efebd1582852fee3
Patch-mainline: v4.19-rc1
References: bsc#1140652

Fixes an error condition reported by checkpatch.pl which is caused by
assigning a variable in an if condition.

(Coly Li: rebase for SLE15 kernel)

Signed-off-by: Florian Schmaus <flo@geekplace.eu>
Signed-off-by: Coly Li <colyli@suse.de>
Signed-off-by: Jens Axboe <axboe@kernel.dk>

---
 drivers/md/bcache/super.c |   20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

--- a/drivers/md/bcache/super.c
+++ b/drivers/md/bcache/super.c
@@ -796,13 +796,14 @@ static int bcache_device_init(struct bca
 	if (idx < 0)
 		return idx;
 
-	if (!(d->bio_split = bioset_create(4, offsetof(struct bbio, bio),
-					   BIOSET_NEED_BVECS |
-					   BIOSET_NEED_RESCUER)) ||
-	    !(d->disk = alloc_disk(BCACHE_MINORS))) {
-		ida_simple_remove(&bcache_device_idx, idx);
-		return -ENOMEM;
-	}
+	d->bio_split = bioset_create(4, offsetof(struct bbio, bio),
+				     BIOSET_NEED_BVECS | BIOSET_NEED_RESCUER);
+	if (!d->bio_split)
+		goto err;
+
+	d->disk = alloc_disk(BCACHE_MINORS);
+	if (!d->disk)
+		goto err;
 
 	set_capacity(d->disk, sectors);
 	snprintf(d->disk->disk_name, DISK_NAME_LEN, "bcache%i", idx);
@@ -836,6 +837,11 @@ static int bcache_device_init(struct bca
 	blk_queue_write_cache(q, true, true);
 
 	return 0;
+
+err:
+	ida_simple_remove(&bcache_device_idx, idx);
+	return -ENOMEM;
+
 }
 
 /* Cached device */
